cms/src/components/content/ContentOverview.vue

```diff
@@ -86,44 +96,116 @@ const titleType = tagTypeString ? tagTypeString : props.docType;
+
+const filters = ref<FilterInput[]>([]);
+
 const showSortOptions = ref(false);
+
 const sortOptionsRef = ref(null);
+
+const filterBtnRef = ref(null);
+const filterInputRef = ref(null);
+
 const sortOption = ref<"title" | "updatedTimeUtc" | "publishDate" | "expiryDate" | undefined>(
     undefined,
 );
 
+const handleResetSorting = () => (queryOptions.value.orderBy = undefined);
+const handleSortByTitle = () => (queryOptions.value.orderBy = "title");
+const handleSortByUpdated = () => (queryOptions.value.orderBy = "updatedTimeUtc");
+const handleSortAscending = () => (queryOptions.value.orderDirection = "asc");
+const handleSortDescending = () => (queryOptions.value.orderDirection = "desc");
+
+const handleSearch = () => {
+    queryOptions.value.search = searchTerm.value;
+    searchTerm.value = "";
+};
+
+const handleNewFilter = (event: Event) => {
+    const target = event.target as HTMLElement;
+    target.classList.add("hidden");
+
+    const filterInput = document.getElementById("filterInput");
+    filterInput?.classList.remove("hidden");
+    filterInput?.focus();
+};
+
+const handleAddNewFilter = () => {
+    const filterBtn = document.getElementById("addFilterBtn");
+    const filterInputEl = document.getElementById("filterInput");
+
+    if (filterInput.value) {
+        const input = filterInput.value.trim().toLowerCase();
+        console.log(input);
+        const newInput = input.split(":");
+        console.log(newInput);
+
+        const newId =
+            filters.value.length > 0
+                ? (Math.max(...filters.value.map((f) => parseInt(f.id))) + 1).toString()
+                : "1";
+
+        const newFilter: FilterInput = {
+            id: newId,
+            filter: newInput[0],
+            by: newInput[1] as Filter,
+        };
+        console.log(newFilter);
+
+        filterInput.value = "";
+        filters.value.push(newFilter);
+    }
+
+    filterBtn?.classList.remove("hidden");
+    filterInputEl?.classList.add("hidden");
+};
+
+const handleFilterDelete = async (event: Event) => {
+    const element = event.target as HTMLElement;
+
+    if (element.tagName === "BUTTON") {
+        const idToDelete = element.id;
+        console.log("Id to delete:", idToDelete);
+        const currentFilters = [...filters.value];
+        filters.value = currentFilters.filter((val) => {
+            console.log("Checking filter with ID:", val.id);
+            return idToDelete !== val.id;
+        });
+
+        await nextTick();
+    }
+
+    console.log(filters.value);
+};
+
 watch(sortOption, () => {
     queryOptions.value.orderBy = sortOption.value;
 });
 
+watch(filters, () => {
+    filters.value.map((filter) => {
+        if (filter.filter == "translation")
+            queryOptions.value.translationStatus = filter.by as
+                | "all"
+                | "translated"
+                | "untranslated"
+                | undefined;
+        console.log(filter);
+    });
+});
+
```

Comment:
### Filter implementation
I am adding filters this way, but it seems that it's not really working